### PR TITLE
Add support for attachments

### LINKED ISSUE #479: Attachments
Are there any plans to support attachments like some of the other libraries does?

https://docs.sentry.io/platforms/javascript/enriching-events/attachments/
https://docs.sentry.io/platforms/native/enriching-events/attachments/
https://docs.sentry.io/platforms/java/enriching-events/attachments/

Looks like it's a separate endpoint for uploading attachments: https://develop.sentry.dev/sdk/envelopes/

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes #479.

### CODE REVIEW

**Comment by whatyouhide** [comment]
@sl0thentr0py ready for review!

File: lib/sentry/envelope.ex [hunk 1]
```diff
@@ -2,37 +2,35 @@ defmodule Sentry.Envelope do
...
+  Creates a new envelope containing the given event and all of its attachments.
   """
-  @spec new([Event.t(), ...]) :: t()
-  def new([%Event{event_id: event_id}] = events) do
+  @spec from_event(Event.t()) :: t()
+  def from_event(%Event{event_id: event_id} = event) do
     %__MODULE__{
       event_id: event_id,
-      items: events
+      items: [event] ++ event.attachments
```
Comment (sl0thentr0py) [inline lib/sentry/envelope.ex:22]: maybe later let's add an `Envelope.add_item` method too when we need other types so that envelope is general enough

**Review by sl0thentr0py — APPROVED** [review]
lgtm 
just added an observation for later

File: lib/sentry/envelope.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/sentry/envelope.ex:22]: Yep we'll probably need something like that down the line!

**Pull request merged**